django_adobesign/backend.py

```diff
@@ -0,0 +1,141 @@
+import logging
+
+from django_anysign import api as django_anysign
+
+from django_adobesign.exceptions import AdobeSignNoMoreSignerException
+
+
+class AdobeSignBackend(django_anysign.SignatureBackend):
+    def __init__(self, adobesign_client, name='AdobeSign', code='adobesign',
+                 url_namespace='anysign'):
+        """Setup.
+
+        Additional keyword arguments are passed to
+        :class:`~django_adobesign.client.AdobeSignCient` constructor, in order
+        to setup :attr:`adobesign client`.
+
+        """
+        super(AdobeSignBackend, self).__init__(
+            name=name,
+            code=code,
+            url_namespace=url_namespace,
+        )
+        self.adobesign_client = adobesign_client
+
+    def get_adobesign_participants(self, signature):
+        """Return list of AdobeSign's Signers for Signature instance.
+
+        Default implementation reads name and email from database.
+
+        """
+        jsonified_signers = []
+        signers = signature.signers.all().order_by('signing_order')
+        for position, signer in enumerate(signers, start=1):
+            signer = self.adobesign_client.jsonify_participant(
+                name=signer.full_name,
+                email=signer.email,
+                order=position
+            )
+            jsonified_signers.append(signer)
+        return jsonified_signers
+
+    def create_signature(self, signature, post_sign_redirect_url=None,
+                         post_sign_redirect_delay=0,
+                         send_mail=True, **extra_data):
+        """Register ``signature`` in AdobeSign service, return updated object.
+        This method calls ``save()`` on ``signature`` and ``signer``.
+
+        """
+        document = next(signature.signature_documents())
+
+        # Upload document
+        response = self.adobesign_client.upload_document(document)
+        transient_document_id = response.get('transientDocumentId')
+        result = self.adobesign_client.post_agreement(
+            transient_document_id=transient_document_id,
+            name=str(signature),
+            participants=self.get_adobesign_participants(signature),
+            state=signature.state,
```

Comment:
why `signature.state` is passed here ?
you should set an argument `state` to the method `create_signature` with a default value.
`signature.state` enum in EFM do not match with adobe state values.

Reply:
Yes its a mistake, we need to pass one of this argument, and not the state of the signature but for our defense its the same name :) 

state : 
a) DRAFT - to incrementally build the agreement  sending out,
b) AUTHORING - to add/edit form fields in the agreement,
c) IN_PROCESS - to immediately send the agreement